src/nvme/util.c

```diff
@@ -434,6 +434,10 @@ int nvme_get_feature_length(int fid, __u32 cdw11, __u32 *len)
 	case NVME_FEAT_FID_WRITE_PROTECT:
 		*len = 0;
 		break;
+	case NVME_MI_FEAT_CTRL_METADATA:
+	case NVME_MI_FEAT_NS_METADATA:
+		*len = 4096;
```

Comment:
After looking at the rest of the function, it looks like the idea is to define the struct to the right size and just do a sizeof(struct nvme_mi_host_metadata) here.